source/dbref.rst

```diff
@@ -0,0 +1,349 @@
+==================
+Handling of DBRefs
+==================
+
+:Spec Title: Handling of DBRefs
+:Spec Version: 1.0
+:Author: Jeremy Mikola
+:Status: Draft
+:Type: Standards
+:Last Modified: 2021-05-21
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+DBRefs are a convention for expressing a reference to another document as an
+embedded document (i.e. BSON type 0x03). Several drivers provide a model class
+for encoding and/or decoding DBRef documents. This specification will both
+define the structure of a DBRef and provide guidance for implementing model
+classes in drivers that choose to do so.
+
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`__.
+
+This specification presents documents as Extended JSON. Although JSON fields are
+unordered, the order of fields presented herein should be considered pertinent.
+This is especially relevant for the `Test Plan`_.
+
+
+Specification
+=============
+
+
+DBRef Structure
+---------------
+
+A DBRef is an embedded document with the following fields:
+
+- ``$ref``: required string field. Contains the name of the collection where
+  the referenced document resides. This MUST be the first field in the DBRef.
+- ``$id``: required field. Contains the value of the ``_id`` field of the
+  referenced document. This MUST be the second field in the DBRef.
+- ``$db``: optional string field. Contains the name of the database where the
+  referenced document resides. If specified, this MUST be the third field in the
+  DBRef. If omitted, the referenced document is assumed to reside in the same
+  database as the DBRef.
+- Extra, optional fields may follow after ``$id`` or ``$db`` (if specified).
+  There are no inherent restrictions on extra field names; however, older server
+  versions may impose their own restrictions (e.g. no dots or dollars).
+
+DBRefs have no relation to the deprecated DBPointer BSON type (i.e. type 0x0C).
+
+
+Examples of Valid DBRefs
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following examples are all valid DBRefs:
```

Comment:
I was using Extended JSON here purely to represent the ObjectId value for `$oid`. I'm happy to clarify that if you think it's helpful. Otherwise I can just change it to `ObjectId("...")` if that would avoid any confusion.

Reply:
Added a clarification to META:

> This specification presents documents as Extended JSON for readability and expressing special types (e.g. ObjectId).

Also clarified in the Test Plan section that drivers should not use an Extended JSON parser but rather construct the corresponding BSON using their native types directly.